kdl_parser/CMakeLists.txt

```diff
@@ -1,55 +1,66 @@
-cmake_minimum_required(VERSION 2.8.3)
+cmake_minimum_required(VERSION 3.5)
 
 project(kdl_parser)
 
-find_package(Boost REQUIRED)
-include_directories(${Boost_INCLUDE_DIR})
-
-find_package(catkin REQUIRED
-  COMPONENTS roscpp rosconsole urdf cmake_modules
-)
+find_package(ament_cmake_ros REQUIRED)
 find_package(orocos_kdl REQUIRED)
+find_package(tinyxml_vendor REQUIRED)
 find_package(TinyXML REQUIRED)
+find_package(urdf REQUIRED)
+find_package(urdfdom_headers REQUIRED)
+find_library(KDL_LIBRARY REQUIRED
+  NAMES orocos-kdl
+  HINTS ${orocos_kdl_LIBRARY_DIRS})
 
-include_directories(include ${orocos_kdl_INCLUDE_DIRS} ${TinyXML_INCLUDE_DIRS} ${catkin_INCLUDE_DIRS})
-
-link_directories(${catkin_LIBRARY_DIRS})
-link_directories(${Boost_LIBRARY_DIRS})
+include_directories(
+  include
+  ${orocos_kdl_INCLUDE_DIRS}
+  ${urdf_INCLUDE_DIRS}
+  ${urdfdom_headers_INCLUDE_DIRS}
+  ${TinyXML_INCLUDE_DIRS})
 
-add_compile_options(-std=c++11)
+link_directories(
+  ${orocos_kdl_LIBRARY_DIRS}
+  ${urdf_LIBRARY_DIRS})
 
-catkin_package(
-  LIBRARIES ${PROJECT_NAME} ${orocos_kdl_LIBRARIES}
-  INCLUDE_DIRS include
-  CATKIN_DEPENDS roscpp rosconsole urdf
-  DEPENDS orocos_kdl TinyXML
-)
+# Default to C++14
+if(NOT CMAKE_CXX_STANDARD)
+  set(CMAKE_CXX_STANDARD 14)
+endif()
 
-add_library(${PROJECT_NAME} src/kdl_parser.cpp)
-target_link_libraries(${PROJECT_NAME}
-  ${TinyXML_LIBRARIES} ${orocos_kdl_LIBRARIES} ${catkin_LIBRARIES}
-)
+add_library(
+  ${PROJECT_NAME}
+  src/kdl_parser.cpp)
+target_link_libraries(
+  ${PROJECT_NAME}
+  ${TinyXML_LIBRARIES}
+  ${orocos_kdl_LIBRARIES}
+  ${urdf_LIBRARIES})
 
 if(WIN32)
   target_compile_definitions(${PROJECT_NAME} PRIVATE "KDL_PARSER_BUILDING_DLL")
 endif()
 
-add_executable(check_kdl_parser src/check_kdl_parser.cpp )
-target_link_libraries(check_kdl_parser ${PROJECT_NAME})
+install(
+  TARGETS ${PROJECT_NAME}
+  LIBRARY DESTINATION lib
+  ARCHIVE DESTINATION lib
+  RUNTIME DESTINATION bin)
 
-if(CATKIN_ENABLE_TESTING)
-  find_package(catkin REQUIRED COMPONENTS rostest)
-  add_rostest_gtest(test_kdl_parser test/test_kdl_parser.launch test/test_kdl_parser.cpp )
-  target_link_libraries(test_kdl_parser ${PROJECT_NAME})
+install(
+  DIRECTORY include/${PROJECT_NAME}/
+  DESTINATION include/${PROJECT_NAME})
 
-  add_rostest_gtest(test_inertia_rpy test/test_inertia_rpy.launch test/test_inertia_rpy.cpp )
-  target_link_libraries(test_inertia_rpy ${PROJECT_NAME})
+if(BUILD_TESTING)
+  find_package(ament_lint_auto REQUIRED)
+  # FIXME(clalancette): we have to disable copyright checking until
```

Comment:
We generally use only "TODO" in the codebase as this is the only one checked by linters so I'd suggest switching this to a TODO

Reply:
Switched to TODO.